source/core/WorldBase.hpp

```diff
@@ -28,20 +28,25 @@ class WorldBase {
 public:
   static constexpr size_t npos = static_cast<size_t>(-1);
 
+  /// Helper function that is run whenever a new agent is created.
+  /// @note Override this function to provide agents with actions or other
+  /// setup.
+  virtual void ConfigAgent(AgentBase & /* agent */) const {}
```

Comment:
Const version was needed as trackingAgent and PathAgent need to reinitialize in case a new path is added or the path wasn't given upon construction. 
We did not want to mess up other groups by removing the non-const version.